### ISSUE TITLE
Azure SQL Server delete behavior in transaction

### ISSUE THREAD

**Issue by shubham1172** [dapr/components-contrib#1517]
# Problem

Say a transaction with only one delete operation is executed on SQL server:
```
delete k1
```
If `k1` is not present in the DB already, this transaction will fail with the error message: `delete affected only 0 rows, expected 1`. 

Is this an expected behavior? 

It is not reproducible with other state stores. If the item is not available, they do not error out.

# Proposal

If the above behavior is unexpected, I see two easy ways to fix this:

1. We can use individual deletes instead of bulk delete in our transactions (performance will take a hit), OR
1. Ignore this type of error in `executeMulti` during bulk delete invocation (should be better)

# Background

Transactions in Azure SQL Server use the bulk delete function to execute all delete operations at once.
```go
func (s *SQLServer) executeMulti(sets []state.SetRequest, deletes []state.DeleteRequest) error {
        // ...
	if len(deletes) > 0 {
		err = s.executeBulkDelete(tx, deletes)
		if err != nil {
			tx.Rollback()

			return err
		}
	}
        // ...
}
```

This is the important part from the bulk delete snippet:
```go
func (s *SQLServer) executeBulkDelete(db dbExecutor, req []state.DeleteRequest) error {
	// ...
	rows, err := res.RowsAffected()
	if err != nil {
		return err
	}

	if int(rows) != len(req) {
		err = fmt.Errorf("delete affected only %d rows, expected %d", rows, len(req))

		return err
	}
        // ...
}
```

**Comment by shubham1172** [comment]
Added a PR for fix with 2nd proposal.

**Comment by shubham1172** [comment]
Closed by #1516
